File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptArrayOptFloatValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptArrayOptFloatInValue.OptArrayOptFloatValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptArrayOptFloatValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptArrayOptFloatValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                Float[].class,
                OptArrayOptFloatValue.class.getDeclaredField("optArrayOptFloat").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                Float[].class,
                OptArrayOptFloatValue.class.getMethod("getOptArrayOptFloat").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptArrayOptFloatValue.class,
                OptArrayOptFloatValue.class.getMethod("setOptArrayOptFloat", Float[].class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptArrayOptFloatValue().getOptArrayOptFloat());
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptArrayOptFloatValue.scala
package com.dslplatform.ocd.values

import OptArrayOptFloatInValue.OptArrayOptFloatValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptArrayOptFloatValue extends SpecExtensions {
  def `Test property field type: Option[Array[Option[Float]]]`() =
    checkType(
      typeOf[Option[Array[Option[Float]]]]
    , typeOf[OptArrayOptFloatValue].member("optArrayOptFloat": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptArrayOptFloatValue().optArrayOptFloat === None)
  }
}
